### ISSUE TITLE
Implement BatchNormalization

### ISSUE THREAD

**Issue by thushv89** [thushv89/AdaCNN#19]

**Comment by thushv89** [comment]
Did not work in non-stationary settings. For batch normalization to work well the dataset needs to be uniform
